jax/interpreters/xla.py

```diff
@@ -989,7 +989,7 @@ def _tuple_output(*args, **kwargs):
   ans = yield args, kwargs
   yield (ans,)
 
-def lower_fun(fun, multiple_results, parallel=False, with_avals=False):
+def lower_fun(fun, multiple_results, parallel=False, with_avals=False, backend=None):
```

Comment:
In a follow-up PR, I'd be tempted to make `backend` mandatory, given that it seems highly likely that anyone not providing a `backend` might hit similar misbehavior.

Reply:
sgtm, wasn't sure how wide-spread this buggy behavior would be. I'll send a follow-up.

Reply:
Well, I'd guess it's rare! But this way we need never hit this bug again...